### PR TITLE
[sonic-config-engine] Parse AutoNegotiation element from LinkMetadata section of minigraph file

### PR DESCRIPTION
**Pull request by jleveque**
#### Why I did it

To enable parsing the `AutoNegotiation` element from the LinkMetadata section of minigraph file

#### How I did it

Parse the value `AutoNegotiation` element from the `LinkMetadata` section of minigraph file. If the element is present, an `autoneg` key will be added to the port in the `PORT` table of Config DB with a value of either `off` or `on`

If an `autoneg` value is present in port_config.ini, the value from the minigraph will take precedence, overriding that value.

Also remove `AutoNegotiation` and `EnableAutoNegotiation` elements from the `DeviceInfo` section, as we will use this data in the `LinkMetadata` section to determine whether to enable auto-negotiation for a port.

#### How to verify it

Added unit tests to validate the behavior described above

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

### CODE REVIEW

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -1318,6 +1323,11 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
         if port.get('speed') == '100000' and linkmetas.get(alias, {}).get('FECDisabled', '').lower() != 'true':
             port['fec'] = 'rs'
 
+        # If AutoNegotiation is available in the minigraph, we override any value we may have received from port_config.ini
+        autoneg = linkmetas.get(alias, {}).get('AutoNegotiation')
+        if autoneg:
+            port['autoneg'] = '1' if autoneg.lower() == 'true' else '0'
```
[the remark this answers is not in the document]
Comment (jleveque) [reply src/sonic-config-engine/minigraph.py:1329]: Thanks for the heads-up! I wasn't aware of this change. As I wrote in my PR descripton above,

>(I would prefer to use on and off, but this retains backward-compatibility with the values obtained from port_config.ini).

However, all of our port_config.ini files still specify `0` or `1`. How should we handle this?

[the remark this answers is not in the document]
Comment (jleveque) [reply src/sonic-config-engine/minigraph.py]: Done in latest commit

**Review by lguohan — APPROVED** [review]

**Comment by daall** [comment]
@jleveque hit a merge conflict on the tests cherry-picking to 202012, can you please submit a PR for 202012

**Comment by jleveque** [comment]
> @jleveque hit a merge conflict on the tests cherry-picking to 202012, can you please submit a PR for 202012

I think it will cherry-pick cleanly after https://github.com/Azure/sonic-buildimage/pull/6901 is cherry-picked.

**Comment by jleveque** [comment]
Opened dedicated PR against 202012 branch [here](https://github.com/Azure/sonic-buildimage/pull/7166). Removing "Request for 202012 branch" label.

**Pull request merged**